core/src/main/java/io/confluent/rest/metrics/MetricsResourceMethodApplicationListener.java

```diff
@@ -218,6 +220,24 @@ public MethodMetrics(ResourceMethod method, PerformanceMetric annotation, Metric
           "The maximum request latency in ms", metricTags);
       this.requestLatencySensor.add(metricName, new Max());
 
+      Percentiles percs = new Percentiles(4 * 100,
+          0.0,
+          100.0,
```

Comment:
how did we pick 100ms for maximum?

Reply:
as I mentioned I just picked an arbitrary number here, hoping we could finalize them in the PR discussion :)

Reply:
The way I would think about this is to pick an absolute maximum upper threshold that we would deem unacceptable for any http endpoint, e.g. 10 seconds, beyond which we don't even care to know how bad things are, while still giving us the granularity we need in the nominal path (e.g. 100 buckets would give us 100ms granularity in this case). For low latency services, maybe 5s max, 50ms granularity is better.
The key thing to keep in mind is that once we set a max, e.g. 5s, then a 5s 99th percentile is not that meaningful anymore, it just tells us that 99th percent of the request are 5s *or more*, the real 99th percentile could be 30s for all we know.

Reply:
I think 10s should cover pretty much all CP components at the moment. The only question would be if we 100 ms or 50 ms granularity.

Reply:
rest proxy has a small number of request paths where high latencies are desirable in some cases. in particular, consumers blocking when there is no data is a good thing, which results in high request latencies.

worth noting since it might make certain percentiles hard to use in that case, e.g. if you have sufficient traffic that consumes low-flow topics.

not sure that specific case should keep us from providing sane defaults for others since i'm not sure how to sanely take advantage of aggregate latency metrics in that case, but just wanted to point out that there is at least one case where the settings mentioned above might not match expectations.